crates/rooch-rpc-api/src/jsonrpc_types/address.rs

```diff
@@ -63,69 +63,88 @@ impl From<RoochAddressView> for AccountAddress {
 }
 
 #[derive(Debug, Clone)]
-pub struct RoochOrBitcoinAddress {
+pub struct UnitedAddress {
     pub rooch_address: RoochAddress,
     pub bitcoin_address: Option<BitcoinAddress>,
+    pub nostr_public_key: Option<NostrPublicKey>,
 }
 
-pub type RoochOrBitcoinAddressView = StrView<RoochOrBitcoinAddress>;
+pub type UnitedAddressView = StrView<UnitedAddress>;
 
-impl std::fmt::Display for RoochOrBitcoinAddressView {
+impl std::fmt::Display for UnitedAddressView {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        if self.0.bitcoin_address.is_some() {
-            return write!(f, "{}", self.0.bitcoin_address.as_ref().unwrap());
+        if let Some(bitcoin_address) = &self.0.bitcoin_address {
+            return write!(f, "{}", bitcoin_address);
+        }
+        if let Some(nostr_public_key) = &self.0.nostr_public_key {
+            return write!(f, "{}", nostr_public_key);
         }
         write!(f, "{}", self.0.rooch_address)
     }
 }
 
-impl FromStr for RoochOrBitcoinAddressView {
+impl FromStr for UnitedAddressView {
     type Err = anyhow::Error;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match RoochAddress::from_str(s) {
-            Ok(rooch_address) => Ok(StrView(RoochOrBitcoinAddress {
+            Ok(rooch_address) => Ok(StrView(UnitedAddress {
                 rooch_address,
                 bitcoin_address: None,
+                nostr_public_key: None,
             })),
             Err(_) => {
                 let bitcoin_address = BitcoinAddress::from_str(s)?;
-                Ok(StrView(RoochOrBitcoinAddress {
+                let nostr_public_key = NostrPublicKey::from_str(s)?;
```

Comment:
If the str is a nostr address, `BitcoinAddress::from_str(s)?` will raise an error.

Reply:
If it is a Nostr address, it can be convert to Bitcoin address, then to rooch address.

Reply:
The NostrPublicKey's `from_bech32` implemention is not correct.

https://github.com/rooch-network/rooch/blob/be1640ca34c2bd8f071764ffe4614ef9ed398ba6/crates/rooch-types/src/address.rs#L868-L881

NostrPublicKey from_bech32 should return `NostrPublicKey`

Reply:
> The NostrPublicKey's `from_bech32` implemention is not correct.
> 
> https://github.com/rooch-network/rooch/blob/be1640ca34c2bd8f071764ffe4614ef9ed398ba6/crates/rooch-types/src/address.rs#L868-L881
> 
> NostrPublicKey from_bech32 should return `NostrPublicKey`

The `FromBech32`, it is originally for RoochAddress for the type NostrPublicKey to be used for converting to a RoochAddress.

Reply:
There's `impl FromBech32 for XOnlyPublicKey`. The `from_bech32` from NostrPublicKey is for converting to RoochAddress.

Reply:
`XOnlyPublicKey::from_bech32` could be applied for this scenario.

Reply:
But if we could convert bech32 string from NostrPublicKey to BitcoinAddress and then to RoochAddress, the NostrPublicKey's from_bech32 implemention could be removed.

Reply:
> But if we could convert bech32 string from NostrPublicKey to BitcoinAddress and then to RoochAddress, the NostrPublicKey's `from_bech32` implemention could be removed.

We just need to figure out steps for converting from NostrPublicKey to BitcoinAddress and remove the implementation for converting from NostrPublicKey to RoochAddress as BitcoinAddress could convert to RoochAddress.

Reply:
> But if we could convert bech32 string from NostrPublicKey to BitcoinAddress and then to RoochAddress, the NostrPublicKey's `from_bech32` implemention could be removed.

NostrPublicKey also need a from_str, from_str=from_bech32

Reply:
> > But if we could convert bech32 string from NostrPublicKey to BitcoinAddress and then to RoochAddress, the NostrPublicKey's `from_bech32` implemention could be removed.
> 
> NostrPublicKey also need a from_str, from_str=from_bech32

Added.